### PR TITLE
Feature/web add applications representation review page

### PR DESCRIPTION
**Pull request by emil-placheta**
## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/web/src/server/applications/applications.router.js [hunk 1]
```diff
@@ -28,8 +28,7 @@ router.use(locals.registerLocals);
 router.use('/search-results', guards.assertDomainTypeExists, applicationsSearchRouter);
 
 router.use('/create-new-case', guards.assertDomainTypeExists, applicationsCreateRouter);
-
-router.use('/case', guards.assertDomainTypeExists, applicationsCaseRouter);
+router.use('/case', applicationsCaseRouter);
```
Comment (jonathangoulding) [inline apps/web/src/server/applications/applications.router.js:31]: guard needs to be put back

File: apps/web/src/styles/_shame.scss [hunk 2]
```diff
@@ -2,3 +2,7 @@
 	position: sticky;
 	top: 0;
 }
+
```
Comment (jonathangoulding) [inline apps/web/src/styles/_shame.scss:5]: remove

File: apps/web/src/server/views/applications/representations/representations-table.component.njk [hunk 3]
```diff
@@ -19,7 +19,7 @@
 				<td class="govuk-table__cell">{{ reps.redacted}}</td>
 				{{ statusTag(reps.status, loop.index) }}
 				<td class="govuk-table__cell">
-					<a href="#"> Review </a>
+					<a href="/applications-service/case/{{caseId}}/relevant-representations/{{reps.id}}/representation-details"> Review </a>
```
Comment (jonathangoulding) [inline apps/web/src/server/views/applications/representations/representations-table.component.njk:22]: might be good to build this in the view model ?

File: apps/web/src/server/views/applications/representations/representation-details/representation-details.njk [hunk 4]
```diff
@@ -0,0 +1,398 @@
+{% extends "applications/layouts/applications.layout.njk" %}
```
Comment (jonathangoulding) [inline apps/web/src/server/views/applications/representations/representation-details/representation-details.njk:1]: make some grouped components out of this to break it down.

File: apps/web/src/server/applications/case/representations/representation-details/application-representation-details.view-model.js [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+const getAttachmentsData = (data) => {
+	return data;
+};
+
+/**
+ * @param {Representation} representation
+ * @returns {object}
+ */
+export const getRepresentationDetailsViewModel = (representation) => {
+	const viewData = {
```
Comment (jonathangoulding) [inline apps/web/src/server/applications/case/representations/representation-details/application-representation-details.view-model.js:142]: can be returned with ({}). might be cleaner ? 

No need to assign the var.

File: apps/web/src/server/applications/case/representations/representation-details/application-representation-details.view-model.js [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+/**
+ *
+ * @param {*} data
+ * @returns {object}
+ */
+const getWorkflowData = (data) => {
+	return data;
+};
+
+// placeholder function
```
Comment (jonathangoulding) [inline apps/web/src/server/applications/case/representations/representation-details/application-representation-details.view-model.js:127]: i would say don't have placeholder funcs. 

What if the work never gets done ? it's dead code.

File: apps/web/src/server/applications/case/representations/representation-details/application-representation-details.view-model.js [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+	return `${text.slice(0, Math.max(0, lastSpaceIndex))}...`;
+};
+
+/**
+ *
+ * @param {Representation} representation
+ * @returns {object}
+ */
+const getRepresentationData = (representation) => {
```
Comment (jonathangoulding) [inline apps/web/src/server/applications/case/representations/representation-details/application-representation-details.view-model.js:96]: prefer destructuring  to better scope the vars needed.

File: apps/web/src/server/applications/case/representations/representation-details/application-representation-details.view-model.js [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+ */
+const formatDate = (date) => format(new Date(date), 'dd MMM yyyy');
+
+/**
+ *
+ * @param {string} text
+ * @param {number} maxLength
+ * @returns {string}
+ */
+const createExcerpt = (text, maxLength) => {
```
Comment (jonathangoulding) [inline apps/web/src/server/applications/case/representations/representation-details/application-representation-details.view-model.js:85]: could be more descriptive ? 

could also be one line or two funcs.

File: apps/web/src/server/applications/case/representations/representation-details/application-representation-details.view-model.js [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+ */
+
+/**
+ * @param {object} representation
+ * @param {Contact[]} representation.contacts
+ * @param {string} type
+ * @returns {object}
+ */
+const getContactDetailsByContactType = ({ contacts }, type) => {
+	let res = {};
```
Comment (jonathangoulding) [inline apps/web/src/server/applications/case/representations/representation-details/application-representation-details.view-model.js:44]: use response if following  the response object pattern. 

Mainly in express to ignore the mistake of thinkings it's req/res

File: apps/web/src/server/applications/case/representations/representation-details/application-representation-details.view-model.js [hunk 10]
```diff
@@ -0,0 +1,151 @@
+import { format } from 'date-fns';
+
+/**
```
Comment (jonathangoulding) [inline apps/web/src/server/applications/case/representations/representation-details/application-representation-details.view-model.js:3]: put these as a type def file in top level representations ?

File: apps/web/src/server/applications/case/representations/representation-details/__tests__/application-representation-details.view-model.unit.test.js [hunk 11]
```diff
@@ -0,0 +1,152 @@
+import { representationDetailsFixture } from '../__fixtures__/representation-details.fixture.js';
+import { getRepresentationDetailsViewModel } from '../application-representation-details.view-model.js';
+
+describe('application representation details view-models', () => {
+	describe('#representationsVieModel', () => {
+		it('should return representation details mapped to the view model', () => {
+			const response = getRepresentationDetailsViewModel(representationDetailsFixture);
+
+			expect(response).toEqual({
```
Comment (jonathangoulding) [inline apps/web/src/server/applications/case/representations/representation-details/__tests__/application-representation-details.view-model.unit.test.js:9]: assert the sub keys not the entire obeject.

EG. 
```
it('should return the agent data', () => {
expect(response.agentData).toEqual({})
})
```

File: apps/web/src/server/applications/case/representations/representation-details/__tests__/application-representation-details.test.js [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+				postcode: 'MD21 5XY'
+			}
+		}
+	],
+	attachments: [],
+	redactedBy: {
+		azureReference: 1
+	}
+};
+// import { representationsFixture } from '../../__fixtures__/representations.fixture.js';
```
Comment (jonathangoulding) [inline apps/web/src/server/applications/case/representations/representation-details/__tests__/application-representation-details.test.js:64]: remove commented out code.

File: apps/web/src/server/applications/case/representations/representation-details/__tests__/application-representation-details.test.js [hunk 13]
```diff
@@ -0,0 +1,97 @@
+import { parseHtml } from '@pins/platform';
+import nock from 'nock';
+import supertest from 'supertest';
+import { createTestEnvironment } from '../../../../../../../testing/index.js';
+
+const { app, installMockApi, teardown } = createTestEnvironment();
+const request = supertest(app);
+
+// Arrange
+const mockAPIResponse = {
```
Comment (jonathangoulding) [inline apps/web/src/server/applications/case/representations/representation-details/__tests__/application-representation-details.test.js:10]: put in a fixture. 

Could look ad braking these key values into small chunks.

File: apps/web/src/server/views/applications/representations/representation-details/representation-details.njk (hunk 4 above)
Comment (emil-placheta) [reply apps/web/src/server/views/applications/representations/representation-details/representation-details.njk:1]: -break down *my* component

File: apps/web/src/server/applications/case/representations/representation-details/application-representation-details.view-model.js (hunk 6 above)
Comment (emil-placheta) [reply apps/web/src/server/applications/case/representations/representation-details/application-representation-details.view-model.js:127]: -remove placeholders

**Comment by emil-placheta** [comment]
fails linting on `npm ci` step, looking into what's happening

File: apps/web/src/server/applications/case/representations/representation-details/__tests__/__snapshots__/application-representation-details.test.js.snap [hunk 14]
```diff
@@ -0,0 +1,178 @@
...
+                    <dd class=\\"govuk-summary-list__value\\">
+                        <div id=\\"representationText\\" class=\\"govuk-!-margin-bottom-2\\">
+                            <noscript>Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo
+                                ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis
+                                dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies
+                                nec, pellentesque eu, pretium quis, sem. Nulla con.</noscript>
+                        </div>
+                        <p class=\\"govuk-body govuk-!-margin-bottom-4\\"><a href=\\"#\\" id=\\"fullRepToggle\\" hidden class=\\"govuk-link govuk-link--no-visited-state\\">Show full representation</a>
+                        </p>
+                        <script nonce=\\"EdcOUaJ8lczj9tIPO0lPow==\\">
```
Comment (jonathangoulding) [inline apps/web/src/server/applications/case/representations/representation-details/__tests__/__snapshots__/application-representation-details.test.js.snap:120]: Can lift out into the scripts dir.

File: apps/web/src/server/views/applications/representations/representation-details/partials/workflow.njk [hunk 15]
```diff
@@ -0,0 +1,19 @@
+<h3 class="govuk-heading-m">Workflow</h3>
```
Comment (jonathangoulding) [inline apps/web/src/server/views/applications/representations/representation-details/partials/workflow.njk:1]: this should be a macro getting the data required via params.

File: apps/web/src/server/views/applications/representations/representation-details/partials/person-contact.njk [hunk 16]
```diff
@@ -0,0 +1,128 @@
+<h3 class="govuk-heading-m">Contact details</h3>
```
Comment (jonathangoulding) [inline apps/web/src/server/views/applications/representations/representation-details/partials/person-contact.njk:1]: this should be a macro

File: apps/web/src/server/views/applications/representations/representation-details/partials/attachments.njk [hunk 17]
```diff
@@ -0,0 +1,4 @@
+<h3 class="govuk-heading-m">Attachments</h3>
```
Comment (jonathangoulding) [inline apps/web/src/server/views/applications/representations/representation-details/partials/attachments.njk:1]: this should be a macro- or a partial if no data is required.

File: apps/web/src/server/views/applications/representations/representation-details/partials/agent-contact.njk [hunk 18]
```diff
@@ -0,0 +1,115 @@
+<h3 class="govuk-heading-m">Agent contact details</h3>
```
Comment (jonathangoulding) [inline apps/web/src/server/views/applications/representations/representation-details/partials/agent-contact.njk:1]: this should be a macro

File: apps/web/src/server/views/applications/representations/representation-details/partials/workflow.njk (hunk 15 above)
Comment (jonathangoulding) [reply apps/web/src/server/views/applications/representations/representation-details/partials/workflow.njk]: Added to clean up ticket.

**Review by jonathangoulding — APPROVED** [review]

**Pull request merged**